.github/workflows/php-linting.yml

```diff
@@ -0,0 +1,217 @@
+name: PHP linting
+
+on: pull_request
+
+jobs:
+  changed_files:
+    name: detect changed files
+    runs-on: ubuntu-latest
+    outputs:
+      php: ${{ steps.filter.outputs.php }}
+      misc: ${{ steps.filter.outputs.misc }}
+      php_excluded_files: ${{ steps.filter2.outputs.php_excluded_files }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          list-files: json
+          filters: |
+            php:
+              # If any PHP file changed, they need checking.
+              - added|modified:
+                - '**.php'
+            misc:
+              # If composer or phpcs config changed, there may be a new standard.
+              - 'composer.json'
+              - 'composer.lock'
+              - '.phpcs.config.xml'
+              - '.phpcs.xml.dist'
+              # If the excludelist changed, run to ensure newly non-excluded files pass.
+              - 'bin/phpcs-excludelist.txt'
+              # If other files used by this workflow changed, run it to test those changes.
+              - 'bin/parallel-lint.sh'
+              - '.github/files/php-linting-phpcs.xml'
+              - '.github/matchers/phpcs-problem-matcher.json'
+              - '.github/matchers/php-lint-problem-matcher.json'
+              - '.github/workflows/php-linting.yml'
+
+      - id: filter2
+        shell: bash
+        run: |
+          printf '::set-output name=php_excluded_files::%s\n' $(
+          jq -r '.[]' <<'EOF' | grep -Fx -f bin/phpcs-excludelist.txt | jq -nRc '[ inputs ]'
+          ${{ steps.filter.outputs.php_files }}
+          EOF
+          )
+
+  lint:
+    name: PHP lint (${{ matrix.php-versions }})
```

Comment:
Should we keep "PHP Lint" in the name here? Since the UI already shows the action name before, we may not need to display that twice?

![image](https://user-images.githubusercontent.com/426388/100363835-951eff00-2ffd-11eb-8be3-26b576bd7c1a.png)

Reply:
It includes the workflow name in some places, but not others.

![ss1](https://user-images.githubusercontent.com/1030580/100627343-e3314c80-32f4-11eb-94af-427ae5067e71.png)
![ss2](https://user-images.githubusercontent.com/1030580/100627346-e4627980-32f4-11eb-8c1e-51ba778a0c7e.png)

Reply:
Ah, good to know.